front/cypress/e2e/admin/pages_and_menu/create_update_and_view_custom_page.cy.ts

```diff
@@ -0,0 +1,154 @@
+      .each((button) => {
+        cy.wrap(button).click();
+        cy.get('[data-cy="e2e-signed-out-header-section"]')
+          .find('input')
+          .type(headerContent);
+      });
+
+    // update the subheader text
+    cy.get('[data-cy="e2e-signed-out-subheader-section"]').scrollIntoView();
+    cy.get('[data-cy="e2e-signed-out-subheader-section"]')
+      .find('.e2e-localeswitcher')
+      .each((button) => {
+        cy.wrap(button).click();
+        cy.get('[data-cy="e2e-signed-out-subheader-section"]')
+          .find('input')
+          .type(subheaderContent);
+      });
+
+    // enable custom button
+    cy.get('[data-cy="e2e-cta-settings-custom-customized_button"]').click();
+
+    cy.get('[data-cy="e2e-cta-settings-custom-customized_button"]')
+      .find('.e2e-localeswitcher')
+      .each((button) => {
+        cy.wrap(button).click();
+        cy.get('[data-testid="inputMultilocLocaleSwitcher"]')
+          .find('input')
+          .type(ctaContent);
+      });
+
+    // type the url
+    cy.get('#buttonConfigInput').type('https://www.coolwebsite.biz');
+
+    // submit form
+    cy.get('.e2e-submit-wrapper-button').click();
+
+    // scroll to breadcrumbs, go back to main page
+    cy.get(`[data-cy="breadcrumbs-${pageName}"]`).click();
+
+    // to do: test color and opacity
+    // go to top info section edit page
+    cy.get(
+      '[data-cy="e2e-admin-edit-button-top_info_section_enabled"]'
+    ).click();
+
+    // fill out top info section
+    cy.get('[data-cy="e2e-top-info-form"]')
+      .find('.e2e-localeswitcher')
+      .each((button) => {
+        cy.wrap(button).click();
+        // to do- remove locale from id, it doesn't work
+        cy.get('#top_info_section_multiloc-en').type(topInfoContent);
+        cy.wrap(button).get('.notEmpty');
+      });
+
+    // submit
+    cy.get('[data-cy="e2e-top-info-section-submit"]').click();
+
+    // scroll to breadcrumbs, go back to main page
+    cy.get(`[data-cy="breadcrumbs-${pageName}"]`).click();
+
+    // go to attachments section edit page
+    cy.get('[data-cy="e2e-admin-edit-button-files_section_enabled"]').click();
+
+    // drop an example pdf file
+    cy.get('#local_page_files').selectFile('cypress/fixtures/example.pdf');
+
+    // submit
+    cy.get('[data-cy="e2e-attachments-section-submit"]').click();
+
+    // wait for success toast
+    cy.get('[data-testid="feedbackSuccessMessage"');
+
+    //  go back to main page
+    cy.get(`[data-cy="breadcrumbs-${pageName}"]`).click();
+
+    // visit our custom page by slug
+    cy.visit(`/en/pages/${pageName}`);
+
+    // check for presence of image in header and text content
+    cy.get('[data-cy="e2e-header-image-background"]')
+      .should('have.attr', 'src')
+      .should('include', '.png');
+    cy.contains(headerContent);
+    cy.contains(subheaderContent);
+    cy.contains(ctaContent);
+    cy.contains(topInfoContent);
+    cy.contains('example.pdf');
+  });
```

Comment:
I think it would be a good idea to also test deleting the page? Then we truly cover the entire flow.

Reply:
Good call, I was going to add that as an API call but it should be via the user UI. 

I think we're going to merge this one and then delete all of the old `page` code, so there will be a second round of updating the e2e tests for this feature before we merge it entirely. I'll take care of it then :)